internal/cmd/schema-registry/command_schema_create.go

```diff
@@ -113,3 +151,31 @@ func (c *command) schemaCreate(cmd *cobra.Command, _ []string) error {
 	output.Printf(errors.RegisteredSchemaMsg, response.Id)
 	return nil
 }
+
+func readMetadata(metadataPath string) (*srsdk.Metadata, error) {
+	var metadata srsdk.Metadata
+	if metadataPath != "" {
+		metadataBlob, err := os.ReadFile(metadataPath)
+		if err != nil {
+			return nil, err
+		}
+		if err := json.Unmarshal(metadataBlob, &metadata); err != nil {
+			return nil, err
+		}
+	}
+	return &metadata, nil
+}
+
+func readRuleset(rulesetPath string) (*srsdk.RuleSet, error) {
```

Comment:
You could likely combine this with the function above, if you pass the object to be read as a pointer argument.